src/realm/object-store/impl/collection_notifier.cpp

```diff
@@ -51,133 +51,74 @@ CollectionNotifier::get_modification_checker(TransactionChangeInfo const& info,
         };
     }
 
+    // If the table in question has no outgoing links it will be the only entry in `m_related_tables`.
+    // In this case we do not need a `DeepChangeChecker` and check the modifications against the
+    // `ObjectChangeSet` within the `TransactionChangeInfo` for this table directly.
     if (m_related_tables.size() == 1) {
-        auto& object_set = info.tables.find(m_related_tables[0].table_key.value)->second;
+        auto& object_chenge_set = info.tables.find(m_related_tables[0].table_key.value)->second;
         return [&](ObjectChangeSet::ObjectKeyType object_key) {
-            return object_set.modifications_contains(object_key);
+            if (all_callbacks_filtered()) {
```

Comment:
This conditional should be outside of the lambda.

Reply:
Fixed with a70d62050.